server/src/main/kotlin/org/javacs/kt/SourceFiles.kt

```diff
@@ -106,7 +106,13 @@ class SourceFiles(private val sp: SourcePath) {
     private fun readFromDisk(file: Path): SourceVersion? {
         if (!Files.exists(file)) return null
 
-        val content = Files.readAllLines(file).joinToString("\n")
+        var content = ""
+
+        try {
+            content = Files.readAllLines(file).joinToString("\n")
+        } catch(exception:Exception) {
```

Comment:
You probably do not want to catch `Exception` directly, but rather something more specific. Can you figure out which exception precisely is thrown?

Reply:
It was on purpose I was catching just an untyped Exception, because it can be either:
- IOException (probably the culprit in my case)
- SecurityException (could happen)
- OutOfMemoryError   

So by convenience it was just easier to just catch an Exception, rather than trying to catch those 3.

Reply:
Especially with the lack of multi exceptions catching in Kotlin

Reply:
@shoebox I agree that it is more convenient to catch multiple exceptions at once, but, except for `IOException`, these exceptions really should not occur in normal usage and might indicate a deeper issue. Thus the code that causes them should be fixed instead.

Could you provide stacktraces and/or logs for the SecurityException and the OutOfMemoryError you encountered?

Reply:
@fwcd I never encountered it, I just was listing whatever error can be fired, from the method javadoc.

Reply:
What do you think about only catching `IOException`s for now?
```suggestion
        } catch(exception: IOException) {
```

Reply:
Changed 👍